torch/distributed/fsdp/flat_param.py

```diff
@@ -2383,9 +2383,12 @@ def _check_storage_freed(tensor: Tensor):
         )
 
     @staticmethod
-    def _check_storage_allocated(tensor: Tensor):
+    def _check_storage_allocated(tensor: Tensor, fqns: List[str]):
         storage_size: int = tensor._typed_storage()._size()
-        _p_assert(storage_size > 0, "Expects storage to be allocated")
+        _p_assert(
+            storage_size > 0,
+            f"Expects storage to be allocated for FlatParam with  original param fqns {fqns}"
```

Comment:
nit: Adding a newline before the FQNs generally helps with readability I have found.
```suggestion
            f"Expects storage to be allocated for FlatParameter with original parameter FQNs:\n {fqns}"
```